packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -825,108 +1031,131 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
     return result;
   }
 
-  @override
-  bool get sizedByParent => true;
-
   @override
   Size computeDryLayout(BoxConstraints constraints) {
     assert(constraints.biggest.isFinite);
     return constraints.biggest;
   }
 
-  @override
-  void performLayout() {
-    _hasVisualOverflow = false;
+  _ChildIterator _paintOrderIterator() {
```

Comment:
Well looks like `sync*` is not banned you just have to give an explanation? That simplified the implementation a lot.

Reply:
> Well looks like sync* is not banned you just have to give an explanation?

IIRC @dnfield said in https://github.com/flutter/flutter/issues/101227#issuecomment-1247641562 that, "We cannot use sync generators, they create code that is large and slow."

Reply:
See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-syncasync

It generates significantly larger code size than a hand written iterator, and historically was getting used very often where lazy iteration was never used or even desirable.

Ask yourself: will callers of this method _ever_ break iteration early? Will calls to this method typically result in materializing the entire list? If so, don't use iterators, just return a list. Even if not, consider writing your own iterator. I'm told the generator code has improved, but it will never be as compact/efficient as just writing your own.

Reply:
> Well looks like `sync*` is not banned you just have to give an explanation? That simplified the implementation a lot.

https://github.com/dart-lang/sdk/issues/32102